ERCS/erc-1234.md

```diff
@@ -0,0 +1,318 @@
+    /// @param from The address from which tokens are being transferred.
+    /// @param to The address to which tokens are being transferred.
+    /// @param tokenId The ID of the token being transferred. 
+    /// @param amount The amount being transferred.
+    error TransferNotAllowed(address from, address to, uint256 tokenId, uint256 amount);
+
+    /// @notice Takes tokens from one address and transfers them to another.
+    /// @dev Requires specific authorization. Used for regulatory compliance or recovery scenarios.
+    /// @param from The address from which `amount` is taken.
+    /// @param to The address that receives `amount`.
+    /// @param tokenId The ID of the token being transferred.
+    /// @param amount The amount to recall.
+    function recall(address from, address to, uint256 tokenId, uint256 amount) external;
+
+    /// @notice Checks if a transfer is currently possible according to token rules and registered plugins.
+    /// @dev This may involve checks like allowlists, blocklists, transfer limits, etc.
+    /// @param from The address sending tokens.
+    /// @param to The address receiving tokens.
+    /// @param tokenId The ID of the token being transferred.
+    /// @param amount The amount being transferred.
+    /// @return allowed True if the transfer is allowed, false otherwise.
+    function isTransferAllowed(address from, address to, uint256 tokenId, uint256 amount) external view returns (bool allowed);
+
+    /// @notice Checks if a specific user is allowed to interact with the token.
+    /// @dev This is often used for allowlist/KYC checks.
+    /// @param user The address to check.
+    /// @return allowed True if the user is allowed, false otherwise.
+    function isUserAllowed(address user) external view returns (bool allowed);
+
+    /// Derived from EIP-165
+    function supportsInterface(bytes4 interfaceID) external view returns (bool);
+}
+```
+
+*   The `isUserAllowed` and `isTransferAllowed` functions provide views into the implementing contract's compliance and transfer policy logic. The exact implementation of these checks (e.g., internal allowlist, external calls, complex logic) is NOT mandated by this interface standard. However these two functions:
+    - MUST NOT revert. 
+    - MUST NOT change the storage of the contract. 
+    - MAY depend on context (e.g., current timestamp or block.number).
+*   The `recall` function provide a standard mechanism for forcing a transfer from a `from` to a `to` address. This is often known as either "confiscation" / "revocation" or even "recovery" which are all names related to the motivation being the feature itself. The name chose tries to abstract away the motivation and keep a general name. This function:
+    - MUST directly manipulate balances or ownership to transfer the asset from `from` to `to` either by transfering or burning from `from` and minting to `to`. 
```

Comment:
I think there is a typo here `transfering` should be `transferring`

Reply:
Fixed ! Thanks